README.md

```diff
@@ -42,7 +42,346 @@ Or install it yourself as:
+```
+
+### Delivery Details
+
+When a Ruby object includes __Lotus::Mailer__, a developer can customize delivery details, such as `from`, `to` and `subject`.
+
+The `from` field determines the sender of the email. It can accept a string:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  from "noreply@example.com"
+end
+```
+
+The `from` field can also accept a `Proc` to determine its value:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  from -> { customized_sender }
+
+  def customized_sender
+    "user-#{ user.id }@example.com"
+  end
+end
+```
+
+The `to` field represents one or more emails that will be the recipients of the mail.
+
+It can accept a string:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  to "noreply@example.com"
+end
+```
+
+The to field also accept an array of strings:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  to "[noreply1@example.com, noreply2@example.com]"
+end
+```
+
+`To` field can also be configured using a `Proc` to determine its value:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  to -> { customized_recipient }
+
+  def customized_recipient
+    "user-#{ user.id }@example.com"
+  end
+end
+```
+
+The subject of the message is determined by the field `subject`. It can accept a string:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  subject "This is the subject"
+end
+```
+
+The `subject` field can also accept a `Proc` to determine its value:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  subject -> { customized_subject }
+
+  def customized_subject
+    "This is the subject"
+  end
+end
+```
+
+### Delivery method
+
+The global delivery method is defined through the __Lotus::Mailer__ configuration, as:
+
+```ruby
+Lotus::Mailer.configuration do
+  delivery_method :smtp
+end
+```
+
+```ruby
+Lotus::Mailer.configuration do
+  delivery_method :smtp, address: "localhost", port:1025
+end
+```
+
+or using the `delivery` alias:
+
+```ruby
+Lotus::Mailer.configuration do
+  delivery :smtp
+end
+```
+
+It is also possible to configure a custom delivery method with a variable:
+
+```ruby
+MyCustomDeliveryMethod = :smtp
+
+Lotus::Mailer.configure do
+  delivery MyCustomDeliveryMethod, foo:'bar'
+end
+```
+
+The delivery method specified must be compatible with the `Mail` gem, and all `Mail` gem delivery methods are available in __Lotus::Mailer__.
+
+### Multipart mail
+
+Each template associated with the mailer will be used as a part of the email. 
+The `.txt` template will be the `text_part` of the Mail object and the `.html` will correspond to the `html_part`.
+Other formats will be added to the object as attachments.
+
+### Mail Delivery
+
+The `deliver` method is necessary to send the multipart email. It looks at all the associated templates and renders them, as explained above. It instantiates a `mailer` and delivers the email.
+
+```ruby
+InvoiceMailer.deliver
```

Comment:
Should we consider that there is always at least one template? I think the e-mail could go empty or just with an attachment.